### DISCUSSION TITLE
Docker

### DISCUSSION THREAD

**Discussion by avocardio** [OpenBB-finance/OpenBB#184] [Q&A] [answered]
Hello,

I have never heard of docker before, what is it supposed to do here? I finally managed to get it to run but have no idea what it does. I thought it would give a nicely looking interface and red/green colors for up and down percentages no? When I do 

`docker run -it gamestonkterminal:dev`

The program just runs in my cmd prompt as usual. How is this supposed to work? 

Thanks everyone.

**Comment by DidierRLopes** [comment]
Hey,

I'm quite new to docker myself. But the main idea of docker is to be like an independent virtual machine. Like a bubble, where you can set up the machine as you want. In our case our docker performs all the installation necessaries, so when you build it and run it you are effectively running a normal terminal BUT all the installation packages and dependencies have been taking care by us. This way we ensure that all the users have a similar user experience, as docker is cross-platform.

The current issue with docker is that it doesn't output matplotlib figures, so it isn't recommended.

Regarding the colouring, we are gradually improving outputs to add this, for a better user experience.

Hope this helps!

**Reply by avocardio** [reply]
Thanks for the explanation. But the coloring stuff has nothing to do with it right? I thought the code was specially made for the docker environment therefore I thought I HAD to run it in docker to make it show correctly. Because these are the outputs im getting on the normal cmd prompt: 

![ss](https://user-images.githubusercontent.com/74978236/111311116-1f977700-865e-11eb-9c12-097126e20c34.PNG)

Or does it have to do with the native windows compiler that is unable to show color?

**Reply by avocardio** [reply]
Or was this just fixed in #189 ?

**Comment by aia** [comment, marked as answer]
I'll get some Docker going soon.

To run the terminal in Docker, we will need to add flags to disable UI features - plots, browser opens etc.

**Comment by hinxx** [comment]
I was wondering yesterday if providing a pre-built docker image, ready to be ran by end users, is something that could be achieved. This would certainly cut down on the installation issues for users that are not that well versed in the python and stuff.

Not sure how the use of docker image would affect the graphical output (plots, browser), though. Wouldn't that be taken care of by the docker itself, if the image would provide a X11 (Linux windowing) for example?

**Reply by hinxx** [reply]
See #227 for getting the plot to show up.
